### PR TITLE
[animations] OpenContainer demo and polish

### PR DESCRIPTION
**Pull request by goderbauer**
Also adds the scrim to the open container transition.

This is a continuation of https://github.com/flutter/packages/pull/63.

### CODE REVIEW

File: packages/animations/lib/src/open_container.dart [hunk 1]
```diff
@@ -389,6 +389,22 @@ class _OpenContainerRoute extends ModalRoute<void> {
       weight: 8 / 12,
     ),
   ]);
+  final TweenSequence<Color> _scrimFadeInTween = TweenSequence<Color>(
+    <TweenSequenceItem<Color>>[
+      TweenSequenceItem<Color>(
+        tween: ColorTween(begin: Colors.transparent, end: Colors.black54),
+        weight: 4 / 12,
```
Comment (shihaohong) [inline packages/animations/lib/src/open_container.dart]: Is there a reason we use 4 / 12 and 8 / 12 instead of 1 / 3 and 2 / 3 for the weights?

Comment (shihaohong) [reply packages/animations/lib/src/open_container.dart]: Also, looking at the scrim spec, I think the weights are supposed to be 3 / 10 and 7 / 10 here

File: packages/animations/example/lib/container_transition.dart [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+                  title: Text('List item ${index + 1}'),
+                  subtitle: const Text('Secondary text'),
+                );
+              },
+            );
+          }),
+        ],
+      ),
+      floatingActionButton: OpenContainer(
+        openBuilder: (BuildContext context, VoidCallback _) {
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/animations/example/lib/container_transition.dart]: Yeah, we could do that here as well.

File: packages/animations/example/lib/container_transition.dart [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+          return _DetailsPage();
+        },
+        closedElevation: 6.0,
+        closedShape: const RoundedRectangleBorder(
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/animations/example/lib/container_transition.dart]: Done.

File: packages/animations/example/lib/container_transition.dart [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+            borderRadius: BorderRadius.all(Radius.circular(_fabDimension / 2))),
+        closedColor: Theme.of(context).colorScheme.secondary,
+        closedBuilder: (BuildContext context, VoidCallback openContainer) {
+          return SizedBox(
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/animations/example/lib/container_transition.dart]: I don't think there's a great way to do that, unfortunately.

File: packages/animations/example/lib/container_transition.dart [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+            child: Padding(
+              padding: const EdgeInsets.all(70.0),
+              child: Image.asset(
+                'assets/placeholder_image.png',
+              ),
+            ),
+          ),
+          Padding(
+            padding: const EdgeInsets.only(
+              top: 20.0,
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/animations/example/lib/container_transition.dart]: Done.

File: packages/animations/lib/src/open_container.dart (hunk 1 above)
Comment (goderbauer) [reply packages/animations/lib/src/open_container.dart]: Talked offline. This is the fade-through variant, which has this slightly different.

File: packages/animations/example/lib/main.dart [hunk 6]
```diff
@@ -74,7 +74,7 @@ class _TransitionsHomePageState extends State<_TransitionsHomePage> {
                 ),
                 _TransitionListTile(
                   title: 'Container Transform',
-                  subtitle: 'The container acts as a persistent '
+                  subtitle: 'OpenContainer'
                       'element as its dimensions, position, and '
```
Comment (shihaohong) [inline packages/animations/example/lib/main.dart]: Also, remove the rest of the text that comes after 'OpenContainer' :)

Comment (goderbauer) [reply packages/animations/example/lib/main.dart]: lol, I blame this one on github :) 

Done.

**Pull request merged**